lib/ansible/modules/cloud/digital_ocean/digital_ocean_space.py

```diff
@@ -0,0 +1,185 @@
+     - The name of the space.
+    required: true
+  state:
+    description:
+     - Whether the space should be present or absent.
+    default: present
+    choices: ['present', 'absent']
+  region:
+    description:
+    - Region where the space will reside.
+    default: 'nyc3'
+  canned_acl:
+    description:
+    - Canned ACL to apply to the bucket.
+    default: private
+    choices: ['private', 'public-read']
+  access_id:
+    description:
+    - Access ID used for authentication with Digital Ocean Spaces API (Default is DO_ACCESS_KEY_ID).
+    default: ''
+  secret_key:
+    description:
+    - Secret key used for authentication with Digital Ocean Spaces API (Default is DO_SECRET_ACCESS_KEY).
+    default: ''
+extends_documentation_fragment:
+    - digital_ocean.documentation
+    - aws
+notes:
+  - Two environment variables are used for authentication with the Digital Ocean
+    Spaces API.  They are DO_ACCESS_KEY_ID and DO_SECRET_ACCESS_KEY.  You can
+    create this key pair in the Digital Ocean control panel.
+'''
+
+
+EXAMPLES = '''
+- name: create a space
+  digital_ocean_space:
+    name: cool_example_space
+    state: present
+
+- name: remove a space
+  digital_ocean_space:
+    name: cool_example_space
+    state: absent
+
+- name: use API credentials stored in ansible-vault
+  digital_ocean_space:
+    name: cool_example_space
+    state: present
+    access_id: "{{ super_secure_access_id_stored_in_vault }}"
+    secret_key: "{{ super_secure_secret_key_also_stored_in_vault }}"
+
+'''
+
+
+RETURN = ''' # '''
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.digital_ocean import DigitalOceanHelper
+from ansible.module_utils._text import to_native
+
+
+def space_exists(client, module):
+    space_name = module.params.get('name')
+    try:
+        client.head_bucket(Bucket=space_name)
+        return True
+    except:
+        return False
+
+
+def create_space(client, module, result):
+    space_name = module.params.get('name')
+    canned_acl = module.params.get('canned_acl')
+    try:
+        response = client.create_bucket(
+            Bucket=space_name,
+            ACL=canned_acl
+        )
+        result['changed'] = True
+        return result
+    except:
+        module.fail_json(msg="Failed to create space: {0}".format(space_name))
+
+
+def update_space(client, module, result):
+    space_name = module.params.get('name')
+    canned_acl = module.params.get('canned_acl')
+    try:
+        response = client.put_bucket_acl(
+            Bucket=space_name,
+            ACL=canned_acl
+        )
+        result['changed'] = True
+        return result
+    except:
+        module.fail_json(msg="Failed to update space ACL: {0}".format(space_name))
+
+
+def delete_space(client, module, result):
+    space_name = module.params.get('name')
+    try:
+        response = client.delete_bucket(Bucket=space_name)
```

Comment:
You might need to do, if `supports_check_mode=True`

```
if not module.check_mode:
    response = client.delete_bucket(Bucket=space_name)
```
Same for Create and Update.

Reply:
I think I prefer something like this above the try/except:
```
if module.check_mode:
    module.exit_json(changed=True)
```
I think that should be sufficient?